Assets/Scripts/ArenasParametersSideChannel.cs

```
@@ -17,11 +20,54 @@ public ArenasParametersSideChannel()
 
     protected override void OnMessageReceived(IncomingMessage msg)
     {
-        byte[] yamlData = msg.GetRawBytes();
+        byte[] rawData = msg.GetRawBytes();
+        string potentialFilePath = Encoding.UTF8.GetString(rawData);
 
-        /* Create the event args and the YAML data */
-        ArenasParametersEventArgs args = new ArenasParametersEventArgs { arenas_yaml = yamlData, };
-        OnArenasParametersReceived(args);
+        // Check if the received string is a valid file path
+        if (!string.IsNullOrEmpty(potentialFilePath) && File.Exists(potentialFilePath))
+        {
+            // New behavior: read arena config from file
+            byte[] yamlData = ReadArenaConfigFile(potentialFilePath);
+            if (yamlData != null)
+            {
+                ArenasParametersEventArgs args = new ArenasParametersEventArgs { arenas_yaml = yamlData, };
+                OnArenasParametersReceived(args);
+            }
+        }
+        else
+        {
+            // Original behavior: treat raw data as arena content directly
+            ArenasParametersEventArgs args = new ArenasParametersEventArgs { arenas_yaml = rawData, };
+            OnArenasParametersReceived(args);
+        }
+    }
+
+    private byte[] ReadArenaConfigFile(string filePath)
+    {
+        try
+        {
+
+            if (!File.Exists(filePath))
+            {
+                Debug.LogError($"Arena config file not found: {filePath}");
+                return null;
```

What's the user experience for these failures? I know this is the current error handling style in this codebase but I wonder if for things like these it would be better to throw a fatal error

---

from the users perspective there would be a line in the log file saying that the arena could not be loaded but other than that nothing would happen. I agree its probably better to throw a fatal error but worth noting that if we just throw the error then I think AAI would load the default arena and freeze like when you load an invalid config. Although I havent tested it.

---

hmm good point - I guess it wouldn't be the best move to just have the error handling work in one different way here. One day I'll sit down and get the whole codebase!